src/Settings.pm

```diff
@@ -428,14 +428,26 @@ sub loadByHandle {
 	assert(defined $object) if DEBUG;
 
 	my $filename;
-	if ($object->{autoSearch}) {
-		if ($object->{type} == CONTROL_FILE_TYPE) {
-			$filename = _findFileFromFolders($object->{name}, \@controlFolders);
+	my $internalFilename;
+	if ($object->{'autoSearch'}) {
```

Comment:
Why does `$internalFilename` depend on `$object->{autoSearch}`? I don't know the context well enough to understand why this isn't just `my $internalFilename = $object->{internalName} || $object->{name};`.